packages/eas-build-job/src/errors.ts

```diff
@@ -57,8 +59,13 @@ export class UserFacingError extends Error {
 }
 
 export class UnknownError extends UserFacingError {
-  constructor() {
-    super(ErrorCode.UNKNOWN_ERROR, 'Unknown error. See logs for more information.');
+  constructor(buildPhase?: BuildPhase) {
+    super(
+      ErrorCode.UNKNOWN_ERROR,
+      buildPhase
```

Comment:
Maybe also check if the `buildPhase` is in the keys of `buildPhaseDisplayName`, just to cover the case when `buildPhase` is set but the mapping is missing